detekt-gradle-plugin/src/testFixtures/kotlin/io/gitlab/arturbosch/detekt/testkit/Helpers.kt

```diff
@@ -1,6 +1,32 @@
 package io.gitlab.arturbosch.detekt.testkit
 
 import org.gradle.api.Task
+import org.intellij.lang.annotations.Language
 
 fun Task.dependenciesAsNames() = this.taskDependencies.getDependencies(this).map { it.name }
 fun Task.dependenciesAsPaths() = this.taskDependencies.getDependencies(this).map { it.path }
+
+/**
+ * This assumes the usual structure of a Kotlin test file:
+ * ```
+ * class Test {
+ *     fun `test name`() {
+ *         val code = """
+ *             // code
+ *             ${otherCode.reIndent()}
+ *             // more code
+ *             some.block {
+ *                 ${innerCode.reIndent(1)}
+ *             }
+ *         """.trimIndent()
+ *     }
+ * }
+ * ```
+ * In case the call site doesn't look like the above,
+ * set the [baseIndent] parameter to the number of indentation levels until the inside of the `"""`.
+ */
+fun String.reIndent(level: Int = 0, baseIndent: Int = 3): String =
+    this.replaceIndent("    ".repeat(baseIndent + level)).trim()
```

Comment:
Shouldn't it be better to use `trimStart`?

```suggestion
    this.replaceIndent("    ".repeat(baseIndent + level)).trimStart()
```

🥗😂

Reply:
I was thinking about that, but this way, there's definitely no opportunity for hacking, because even if you add multiple lines at start/end, they get removed so the results are consistent.